[DTensor] Fix DeviceMesh

shintaro-iwasaki opened the pull request:
Summary: This Diff fixes some DeviceMesh issues, which blocks internal DTensor integration.  Specifically, when `self.mesh = [2, 3]` while `world_size = 4`, because `unique_mesh_values[-1] == 3`, it takes the first short-cut branch and uses `default_pg`. Let's check the length instead of the last value of `unique_mesh_values`.

Test Plan: CI

Reviewed By: wanchaol

Differential Revision: D44079872

wanchaol reviewed (approved):
lgtm, thanks for the fix!

wanchaol commented:
/easycla

shintaro-iwasaki commented:
/easycla

shintaro-iwasaki commented:
/easycla

shintaro-iwasaki committed:
[DTensor] Fix DeviceMesh (#96861)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96861

This Diff fixes some DeviceMesh issues, which blocks internal DTensor integration.  Specifically, when `self.mesh = [2, 3]` while `world_size = 4`, because `unique_mesh_values[-1] == 3`, it takes the first short-cut branch and uses `default_pg`. Let's check the length instead of the last value of `unique_mesh_values`.

Test Plan: CI

Reviewed By: wanchaol

Differential Revision: D44079872

fbshipit-source-id: add6bf0851577f02d1e7a643324f4c28c88344c1
